server/routes/bookingJourney/selectVisitorsController.ts

```diff
@@ -55,15 +57,62 @@ export default class SelectVisitorsController {
       body('visitorIds')
         .toArray()
         .toInt()
+        // filter out any invalid or duplicate visitorId values
         .customSanitizer((visitorIds: number[], meta) => {
           const req = meta.req as unknown as Express.Request
           const allVisitorIds = req.session.bookingJourney.allVisitors.map(visitor => visitor.personId)
 
           const validVisitorIds = visitorIds.filter(visitorId => allVisitorIds.includes(visitorId))
-          return validVisitorIds
+          const uniqueValidVisitorIds = new Set(validVisitorIds)
+
+          return [...uniqueValidVisitorIds]
         })
         .isArray({ min: 1 })
-        .withMessage('No visitors selected'),
+        .withMessage('No visitors selected')
+        .bail()
+        // validate visitor totals
+        .custom((visitorIds: number[], { req }: Meta & { req: Express.Request }) => {
```

Comment:
Found a neater way here to correctly type `req` in `customSanitizer() | custom()`